x-pack/plugins/security_solution/public/timelines/components/open_timeline/helpers.ts

```diff
@@ -173,29 +173,33 @@ const getTemplateTimelineId = (
   duplicate: boolean,
   targetTimelineType?: TimelineType
 ) => {
-  if (!duplicate) {
-    return timeline.templateTimelineId;
-  }
-
   if (
     targetTimelineType === TimelineType.default &&
     timeline.timelineType === TimelineType.template
   ) {
     return timeline.templateTimelineId;
   }
 
-  // TODO: MOVE TO BACKEND
-  return uuid.v4();
+  return duplicate && timeline.timelineType === TimelineType.template
+    ? // TODO: MOVE TO THE BACKEND
+      uuid.v4()
```

Comment:
The reason why I did it here was to fix templateTimelineId was not available until the request came back.
If we move this to server side may cause update failure if users updating timeline continuously in a short time. Means the follow up request might have sent before the first one come back with a templateTimelineId.